applications/nrf5340_audio/src/modules/sd_card_playback.c

```diff
@@ -0,0 +1,433 @@
+{
+	int ret;
+	uint8_t pcm_b[pcm_frame_size];
+	size_t read_size = pcm_frame_size;
+
+	if (!sd_card_playback_active) {
+		LOG_ERR("SD card playback is not active");
+		return -EACCES;
+	}
+
+	ret = sd_card_playback_ringbuf_read(pcm_b, &read_size);
+	if (ret < 0) {
+		LOG_ERR("Loading data into buffer err: %d", ret);
+		return ret;
+	}
+
+	if (read_size == pcm_frame_size) {
+		ret = pcm_mix(pcm_a, pcm_a_size, pcm_b, pcm_frame_size, B_MONO_INTO_A_STEREO_L);
+		if (ret < 0) {
+			LOG_ERR("Pcm mix err: %d", ret);
+			return ret;
+		}
+	} else {
+		LOG_WRN("Size written (%d) does not equal requested size (%d). Skipping...",
+			read_size, pcm_frame_size);
+	}
+	return 0;
+}
+
+int sd_card_playback_init(void)
+{
+	int ret;
+
+	sd_card_playback_thread_id = k_thread_create(
+		&sd_card_playback_thread_data, sd_card_playback_thread_stack,
+		CONFIG_SD_CARD_PLAYBACK_STACK_SIZE, (k_thread_entry_t)sd_card_playback_thread, NULL,
+		NULL, NULL, K_PRIO_PREEMPT(CONFIG_SD_CARD_PLAYBACK_THREAD_PRIORITY), 0, K_NO_WAIT);
+	ret = k_thread_name_set(sd_card_playback_thread_id, "sd_card_playback");
+	if (ret < 0) {
+		return ret;
+	}
+	return 0;
+}
+
+/* Shell functions */
+static int cmd_play_wav_file(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	char file_loc[MAX_PATH_LEN] = "";
+
+	strcat(file_loc, playback_file_path);
+	strcat(file_loc, argv[1]);
+	ret = sd_card_playback_wav(file_loc, wav_playback_cfg.frame_duration_ms,
+				   wav_playback_cfg.bit_depth, SAMPLE_RATE_48K);
+	if (ret > 0) {
+		shell_error(shell, "WAV playback err: %d", ret);
+		return ret;
+	}
+	return 0;
+}
+
+static int cmd_play_lc3_file(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+
+	if (argc != 2) {
+		shell_error(shell, "Number of args is incorrect");
+		return -EINVAL;
+	}
+	char file_loc[MAX_PATH_LEN] = "";
+
+	strcat(file_loc, playback_file_path);
+	strcat(file_loc, argv[1]);
+	ret = sd_card_playback_lc3(file_loc);
+	if (ret < 0) {
+		shell_error(shell, "LC3 playback err: %d", ret);
+		return ret;
+	}
+	return 0;
+}
+
+static int cmd_change_dir(const struct shell *shell, size_t argc, char **argv)
+{
+	/* Remember to change this to make sure you dont navigate to a nonexistent dir */
+	if (argv[1][0] == '/') {
+		playback_file_path[0] = '\0';
+		shell_print(shell, "Current directory: root");
+	} else {
+		strcat(playback_file_path, argv[1]);
+		strcat(playback_file_path, "/");
+		shell_print(shell, "Current directory: %s", playback_file_path);
+	}
+	return 0;
+}
+
+static int cmd_list_files(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	char buf[LIST_FILES_BUF_SIZE];
+	size_t buf_size = LIST_FILES_BUF_SIZE;
+
+	ret = sd_card_list_files(playback_file_path, buf, &buf_size);
+	if (ret < 0) {
+		shell_error(shell, "List files err: %d", ret);
+		return ret;
+	}
+	shell_print(shell, "%s", buf);
+	return 0;
+}
+
+/* Creating subcommands (level 1 command) array for command "demo". */
```

Comment:
Hmm, what does these two "command "demo"" comments mean? Maybe this is copied from button_handler.c?